### PR TITLE
T-55 Tests

### PR DESCRIPTION
**Pull request by DashaDrahun**
added tests

### CODE REVIEW

File: PortfolioKeeper.Tests/BusinessLogic/Mappers/AssetValuationMapperTests.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                CreatedOn = DateTime.UtcNow,
+                CurrencyId = (byte)CurrencyValue.BYN,
+                Value = price
+            };
+
+            // act
+            var result = underTest.Map(asset, price);
+
+            // assert
+            result.Should().BeEquivalentTo(expectedResult, options => options
```
Comment (goremukin) [inline PortfolioKeeper.Tests/BusinessLogic/Mappers/AssetValuationMapperTests.cs]: `result.Should().BeEquivalentTo(expectedResult, options => options.Excluding(f => f.CreatedOn));`
or
```
result.Should().BeEquivalentTo(expectedResult, options => 
                options.Excluding(f => f.CreatedOn));
```

File: PortfolioKeeper.Tests/BusinessLogic/Services/AssetValuationServiceTests.cs [hunk 2]
```diff
@@ -115,5 +115,47 @@ public void Get_When_TimeRange_Not_Valid_Should_Throw_ArgumentException()
...
+        [TestCase((byte)AssetTypeValue.UsdAccount)]
+        [TestCase((byte)AssetTypeValue.BynAccount)]
+        public void Create(byte assetTypeId)
+        {
+            // arrange
+            var asset = new Asset { AssetTypeId = assetTypeId };
+            var valuation = new AssetValuation();
+            string valueStr = "100";
+            decimal valueDec = 100;
+            helper.Setup(f => f.ParsePrice(valueStr)).Returns(valueDec);
```
Comment (goremukin) [inline PortfolioKeeper.Tests/BusinessLogic/Services/AssetValuationServiceTests.cs]: Empty line before methods, please

File: PortfolioKeeper.Tests/BusinessLogic/Helpers/ValuationHelperTests.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            // assert
+            act.Should().Throw<ArgumentException>();
+        }
+
+        [Test]
+        public void ParsePrice_When_Price_With_Point_Delimeter_Should_Return_Decimal()
+        {
+            // arrange
+            string price = "100.5";
+            decimal expectedResult = (decimal)100.5;
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Helpers/ValuationHelperTests.cs]: remove one decimal - 1) var expectedResult = (decimal)100.5; 2) decimal expectedResult = 100.5;

File: PortfolioKeeper.Tests/BusinessLogic/Helpers/ValuationHelperTests.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+            // act
+            var result = underTest.ParsePrice(price);
+
+            // assert
+            result.Should().Be(expectedResult);
+        }
+
+        [Test]
+        public void ParsePrice_When_Price_With_Comma_Delimeter_Should_Return_Decimal()
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Helpers/ValuationHelperTests.cs]: you can unite this with previous method using TestCase - less code, no duplications

File: PortfolioKeeper.Tests/BusinessLogic/Mappers/AssetMapperTests.cs [hunk 5]
```diff
@@ -81,5 +82,54 @@ public void Map_When_Arguments_Are_Valid_Should_Return_Asset()
...
+
+        [Test]
+        public void MapValue_When_Properties_Contain_Value_Should_Return_PropertyValueViewModel()
+        {
+            //arrange
+            var properties = new List<AssetPropertyValueViewModel>
+            {
+                new AssetPropertyValueViewModel
+                {
+                    TypeId = (int) AssetValue.Count
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Mappers/AssetMapperTests.cs]: no need to use enums here. use testId. Same to expected result - over complicated. Use var expectedResult = "3";
then in change  TypeId = (int) AssetValue.Count ->  TypeId = (int) expectedResult;

File: PortfolioKeeper.Tests/BusinessLogic/Services/AssetValuationServiceTests.cs (hunk 2 above)
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Services/AssetValuationServiceTests.cs]: add line

File: PortfolioKeeper.Tests/BusinessLogic/Services/AssetValuationServiceTests.cs [hunk 6]
```diff
@@ -115,5 +115,47 @@ public void Get_When_TimeRange_Not_Valid_Should_Throw_ArgumentException()
...
+            act.Should().Throw<ArgumentNullException>();
+        }
+
+        [TestCase((byte)AssetTypeValue.GoldAssert)]
+        [TestCase((byte)AssetTypeValue.PlatinumAssert)]
+        [TestCase((byte)AssetTypeValue.SilverAssert)]
+        [TestCase((byte)AssetTypeValue.EurAccount)]
+        [TestCase((byte)AssetTypeValue.UsdAccount)]
+        [TestCase((byte)AssetTypeValue.BynAccount)]
+        public void Create(byte assetTypeId)
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Services/AssetValuationServiceTests.cs]: rename? add description what you are testing here

File: PortfolioKeeper/BusinessLogic/Mappers/AssetValuationMapper.cs [hunk 7]
```diff
@@ -10,6 +10,8 @@ public AssetValuation Map(Asset asset, decimal price)
         {
             if (asset == null)
                 throw new ArgumentNullException(nameof(asset));
+            if (price <= 0)
+                throw new ArgumentException("Price can't be <= 0");
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Mappers/AssetValuationMapper.cs]: argumentoutofrangeexception would be better here

File: PortfolioKeeper.Tests/BusinessLogic/Mappers/AssetMapperTests.cs (hunk 5 above)
Comment (DashaDrahun) [reply PortfolioKeeper.Tests/BusinessLogic/Mappers/AssetMapperTests.cs]: I test not typeId in this case but value. I've got rid of enums in this test, see the last commit

File: PortfolioKeeper.Tests/BusinessLogic/Mappers/AssetMapperTests.cs [hunk 8]
```diff
@@ -91,10 +91,11 @@ public void MapValue_When_Properties_Contain_Value_Should_Return_PropertyValueVi
             {
                 new AssetPropertyValueViewModel
                 {
-                    TypeId = (int) AssetValue.Count
+                    TypeId = 2,
+                    Value = "value"
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Mappers/AssetMapperTests.cs]: create constant and use in both places

File: PortfolioKeeper.Tests/BusinessLogic/Services/AssetValuationServiceTests.cs [hunk 9]
```diff
@@ -132,6 +132,7 @@ public void Create_When_Value_Is_Null_Or_Empty_Should_Throw_ArgumentNullExceptio
             act.Should().Throw<ArgumentNullException>();
         }
 
+        // creating valuation depending on asset type and national currency/not national currency mark
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Services/AssetValuationServiceTests.cs]: Rename test method. no need in comments. Create_When_something_do_something- not just Create

File: PortfolioKeeper/BusinessLogic/Mappers/AssetValuationMapper.cs [hunk 10]
```diff
@@ -11,7 +11,7 @@ public AssetValuation Map(Asset asset, decimal price)
             if (asset == null)
                 throw new ArgumentNullException(nameof(asset));
             if (price <= 0)
-                throw new ArgumentException("Price can't be <= 0");
+                throw new ArgumentOutOfRangeException($"Price can not be <= 0");
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Mappers/AssetValuationMapper.cs]: update tests if there some or write new to test ArgumentOutOfRangeException

**Pull request merged**